### PR TITLE
Transmart essentials

### PR DESCRIPTION
**Pull request by gijskant**
Major update: remove all legacy components. Version increase required, possibly 17.2?

### CODE REVIEW

**Comment by ewelinagr** [comment]
There are few more places in the documentation that can be updated, e.g. [README of transmart-rest-api](https://github.com/thehyve/transmart-core/tree/transmart-essentials/transmart-rest-api#authentication-with-oauth) and [installation docs](https://github.com/thehyve/transmart-core/tree/dev/docs#5-start-services).

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
     - DIR=transmart-rest-api; TEST=check
     - DIR=transmart-core-db; INSTALL=":transmart-data:createVars :transmart-data:setupPostgresTest"; TEST=check
     - DIR=transmart-api-server; TEST=check
-    - DIR=transmart-copy; INSTALL=":transmart-data:createVars :transmart-data:setupPostgres"; PREPARE=shadowJar; TEST=check
+    - DIR=transmart-copy; INSTALL=":prepareTestDatabase"; PREPARE=shadowJar; TEST=check
```
Comment (ewelinagr) [inline .travis.yml]: looks like the `before_install` part can be also cleaned up a bit now, e.g. php and gnupg2 checks could be removed, right?

**Comment by ewelinagr** [comment]
https://github.com/thehyve/transmart-core/pull/428/commits/3b47a79764a697eadd67687a6dbe54efa797239b solves the task described here: https://github.com/thehyve/transmart-core/tree/transmart-essentials/transmart-data/test_studies#test-studies
Also the tm_cz user mentioned there is no longer used, right?

File: .travis.yml (hunk 1 above)
Comment (ewelinagr) [reply .travis.yml]: https://github.com/thehyve/transmart-core/blob/transmart-essentials/.travis.yml#L8

Comment (gijskant) [reply .travis.yml]: I am not sure if those can already be removed. I think php still plays a role in bundling the transmart-data artefact, when `gradle publishToMavenLocal` is run.

**Comment by gijskant** [comment]
Indeed, that documentation can updated.

**Comment by gijskant** [comment]
Documentation in transmart-data updated.

File: docs/README.md [hunk 2]
```diff
@@ -90,104 +65,36 @@ keycloak:
...
 ```
-gradle :transmart-server:assemble
+gradle :transmart-api-server:assemble
 ```
-This should create the file `transmart-server/build/libs/transmart-server-17.1-HYVE-6.2-SNAPSHOT.war`.
+This should create the file `transmart-api-server/build/libs/transmart-api-server-17.2-SNAPSHOT.war`.
 Run it in production mode with:
 ```
-java -jar transmart-server/build/libs/transmart-server-17.1-HYVE-6.2-SNAPSHOT.war
+java -jar -Dspring.config.location=/path/to/config.yaml transmart-api-server/build/libs/transmart-server-17.2-SNAPSHOT.war
```
Comment (ewelinagr) [inline docs/README.md]:
```suggestion
java -jar -Dspring.config.location=/path/to/config.yaml transmart-api-server/build/libs/transmart-api-server-17.2-SNAPSHOT.war
```

File: README.md [hunk 3]
```diff
@@ -17,16 +17,15 @@ For details on contributing code changes via pull requests, [see the Contributin
...
+is available.
 
 The database schema is described in the [data model documentation](docs/data-model.md).
 The schema is created using Liquibase at application startup, when it is configured to do so
 (see [Setup database](docs#2-setup-database)). This is only tested with PostgreSQL.
 The [transmart-copy](transmart-copy) tool can be used to load data into the database.
-
 Legacy database definitions and installation instructions are in [transmart-data](transmart-data),
 tested with both PostgreSQL and Oracle databases.
 A data loading tool based on Spring Batch is available as [transmart-batch](transmart-batch).
```
Comment (ewelinagr) [inline README.md]: This sentence could be removed.

**Pull request merged**